### DISCUSSION TITLE
Pydantic validation issue in `0.3.21`

### DISCUSSION THREAD

**Discussion by drupol** [open-webui/open-webui#5506] [General]
# Bug Report

## Installation Method

NixOS

## Environment

- **Open WebUI Version: v0.3.21
- **Operating System:** Linux NixOS
- **Browser (if applicable):** Firefox

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [ ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Open-WebUI should run

## Actual Behavior:

It crashes at startup

## Description

Here's the systemd logs: https://share.riseup.net/#6jpn2ZjSDuHpvQ3bD4aopA

To get it working, I had to add the variable: `PYDANTIC_SKIP_VALIDATING_CORE_SCHEMAS` set to `True`.

I guess the issue comes from: https://github.com/open-webui/open-webui/blob/b64c9d966a6498d4f2ffe12ab1498fab298afb37/backend/open_webui/apps/rag/utils.py#L438 ? But I'm not sure.

**Comment by tjbck** [comment]
Unable to reproduce, let us know your findings!

**Reply by drupol** [reply]
Maybe this is simply because of:

```
sep 18 23:01:19 nixos open-webui[1558999]: /nix/store/zghn5zb1yv7p0zibwy6g67vxfch0x72c-open-webui-0.3.21/lib/python3.12/site-packages/open_webui/apps/rag/main.py:26: LangChainDeprecationWarning: As of langchain-core 0.3.0, LangChain uses pydantic v2 internally. The langchain_core.pydantic_v1 module was a compatibility shim for pydantic v1, and should no longer be used. Please update the code to import from Pydantic directly.
sep 18 23:01:19 nixos open-webui[1558999]: For example, replace imports like: `from langchain_core.pydantic_v1 import BaseModel`
sep 18 23:01:19 nixos open-webui[1558999]: with: `from pydantic import BaseModel`
sep 18 23:01:19 nixos open-webui[1558999]: or the v1 compatibility namespace if you are working in a code base that has not been fully upgraded to pydantic 2 yet.         from pydantic.v1 import BaseModel
sep 18 23:01:19 nixos open-webui[1558999]:   from open_webui.apps.rag.utils import (
```

?

This could be easily updated, don't you think?

**Reply by drupol** [reply]
I can reproduce the issue, even when completely erasing the config and starting from scratch.

Here's my env. variables: https://github.com/drupol/nixos-x260/blob/84fe658df484f53751b67a2b1a97e22742f09b64/modules/system/ai/default.nix#L32

**Reply by tjbck** [reply]
Updated on dev!
